apps/ehr/src/components/AppointmentTableRow.tsx

```diff
@@ -645,6 +654,51 @@ export default function AppointmentTableRow({
     return undefined;
   };
 
+  const handleDischargeButton = async (): Promise<void> => {
+    setDischargeButtonLoading(true);
+    try {
+      if (!encounter || !encounter.id) {
+        throw new Error('Encounter ID is required to change the visit status');
+      }
+      if (!oystehrZambda) {
+        throw new Error('Oystehr Zambda client is not available when changing the visit status');
+      }
+      if (!user) {
+        throw new Error('User is required to change the visit status');
+      }
```

Comment:
Maybe move the validations to handleChangeInPersonVisitStatus to make the client code simpler?